src/megaapi_impl.cpp

```diff
@@ -23026,10 +23026,16 @@ void MegaApiImpl::sendPendingRequests()
         }
         case MegaRequest::TYPE_BACKUP_PUT:
         {
-            BackupType bType = request->getTotalBytes() < INVALID || request->getTotalBytes() > MEDIA_UPLOAD ?
-                               INVALID : static_cast<BackupType>(request->getTotalBytes());
-            
-            if (request->getParentHandle() == UNDEF) // Register a new sync
+            bool isNew = request->getParentHandle() == UNDEF;
+            int backupType = static_cast<int>(request->getTotalBytes());
+            if (backupType < MegaApi::BACKUP_TYPE_CAMERA_UPLOADS || backupType > MegaApi::BACKUP_TYPE_MEDIA_UPLOADS)
```

Comment:
backupType can be MegaApi::BACKUP_TYPE_INVALID in case of update request. This condition will be satisfied and will prematurely break with API_EARGS without giving a chance to update.

Reply:
Good catch! :muscle: